src/guide/ssr/hydration.md

```diff
@@ -0,0 +1,35 @@
+# Client Side Hydration
+
+Hydration refers to the client-side process during which Vue takes over the static HTML sent by the server and turns it into dynamic DOM that can react to client-side data changes.
+
+In `entry-client.js`, we are simply mounting the app with this line:
+
+```js
+app.mount('#app')
+```
+
+Since the server has already rendered the markup, we obviously do not want to throw that away and re-create all the DOM elements. Instead, we want to "hydrate" the static markup and make it interactive.
+
+Vue provides a `createSSRApp` method for use in client-side code (in this case, in our `entry-client.js`) to tell Vue to hydrate the existing static HTML instead of re-creating all the DOM elements.
+
+### Hydration Caveats
+
+In development mode, Vue will assert the client-side generated virtual DOM tree matches the DOM structure rendered from the server. If there is a mismatch, it will bail hydration, discard existing DOM and render from scratch. There will be a warning in the browser console but your site will still work.
+
+However, in production mode, this assertion is disabled for maximum performance, which means that **it is crucial you ensure your client- and server-side DOM trees match**. Otherwise your HTML will remain static and non-reactive.
```

Comment:
I am not sure about this one, does it really fail completely on hydration mismatch? I am not able to test this right now, but I remember that hydration algorithm in Vue 3 was significantly improved so the app doesn't crash completely in case of a mismatch.

Reply:
Good point. I'd need to check this too.

Reply:
@CyberAP I still can see a warning in the console for this but in prod mode, it's the same as in dev: app just renders from scratch without crashing 👍🏻 I'll unify both paragraphs removing a distinction for modes